### PR TITLE
Update Ruby APM documentation

### PR DESCRIPTION
**Pull request by delner**
### What does this PR do?

Updates the Ruby APM documentation with more details and setup information.

### Motivation

Our documentation is out of date and lacking important information.

### Preview link

Not sure how to get one of these? Concerns https://docs.datadoghq.com/tracing/setup/ruby/

### Additional Notes

We need to update some documentation in the `dd-trace-rb` repo to match as well.

Complementary pull request for our `dd-trace-rb` repository: https://github.com/DataDog/dd-trace-rb/pull/388

### CODE REVIEW

File: content/tracing/setup/ruby.md [hunk 1]
```diff
@@ -18,69 +21,215 @@ further_reading:
...
+    ```ruby
+    require 'ddtrace'
+    Datadog.configure do |c|
+      # Configure the tracer here.
+      # Activate integrations, change tracer settings, etc...
+      # By default without additional configuration, nothing will be traced.
+    end
+    ```
+
+3. Add manual instrumentation around your code (see "Manual Instrumentation")
```
Comment (lejulie) [inline content/tracing/setup/ruby.md]: let's link to the section below

File: content/tracing/setup/ruby.md [hunk 2]
```diff
@@ -18,69 +21,215 @@ further_reading:
...
+
+#### Final steps
 
-Next, install the ddtrace gem:
+After setting up, you should see traces appear on the [services page](https://app.datadoghq.com/apm/services) within a few minutes. Read more about [using the APM UI](https://docs.datadoghq.com/tracing/visualization/).
```
Comment (lejulie) [inline content/tracing/setup/ruby.md]: After setting up, your services will appear on the [APM services page](https://app.datadoghq.com/apm/services) within a few minutes. Learn more about [using the APM UI](https://docs.datadoghq.com/tracing/visualization/).

File: content/tracing/setup/ruby.md [hunk 3]
```diff
@@ -18,69 +21,215 @@ further_reading:
 
 ## Installation
 
-To begin tracing applications written in Ruby, first [install and configure the Datadog Agent](/tracing/setup) (see additional documentation for [tracing Docker applications](/tracing/setup/docker/)).
+The following steps will help you quickly start tracing your Ruby application.
+
+#### Setup the Datadog Agent
```
Comment (lejulie) [inline content/tracing/setup/ruby.md]: nit - why do we jump from H2 to H4?

File: content/tracing/setup/ruby.md [hunk 4]
```diff
@@ -18,69 +21,215 @@ further_reading:
...
-Next, install the ddtrace gem:
+After setting up, you should see traces appear on the [services page](https://app.datadoghq.com/apm/services) within a few minutes. Read more about [using the APM UI](https://docs.datadoghq.com/tracing/visualization/).
+
+## Configuration
+
+To activate more advanced features, change tracer behavior, or trace additional code, you must add additional configuration.
+
+#### Manual instrumentation
+
+If you aren't using a supported framework instrumentation, you may want to to manually instrument your code. Adding tracing to your code is easy using the `Datadog.tracer.trace` method, which you can wrap around any Ruby code.
```
Comment (lejulie) [inline content/tracing/setup/ruby.md]: Let's link to the supported framework section.

File: content/tracing/setup/ruby.md [hunk 5]
```diff
@@ -18,69 +21,215 @@ further_reading:
...
-  span.set_tag("my_tag", "my_value")
+# Now write your code naturally, it's traced automatically.
+get '/home' do
+  @posts = Posts.order(created_at: :desc).limit(10)
+  erb :index
 end
 ```
 
-For more examples, see the [RubyDoc Gem documentation](http://gems.datadoghq.com/trace/docs/).
+For list of available integrations, see "Library Compatibility".
```
Comment (lejulie) [inline content/tracing/setup/ruby.md]: Add link

File: content/tracing/setup/ruby.md [hunk 6]
```diff
@@ -18,69 +21,215 @@ further_reading:
...
+end
+```
+
+For more tracer settings, check out the [API documentation](https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#configure-the-tracer).
+
+#### Priority sampling
+
+Priority sampling allows you to configure which traces are most important and should be kept after sampling.
+
+For more details about how to activate and configure priority sampling, check out the [API documentation](https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#priority-sampling).
```
Comment (lejulie) [inline content/tracing/setup/ruby.md]: Priority sampling is disabled by default.  For more details...

File: content/tracing/setup/ruby.md [hunk 7]
```diff
@@ -18,69 +21,215 @@ further_reading:
...
+
+#### Distributed tracing
+
+Distributed tracing allows you to propogate a single trace across multiple services, so you can see performance end-to-end.
```
Comment (lejulie) [inline content/tracing/setup/ruby.md]: "propagate" is misspelled.  Also add that this is disabled by default.

File: content/tracing/setup/ruby.md [hunk 8]
```diff
@@ -18,69 +21,215 @@ further_reading:
...
+
+For more details about how to activate and configure distributed tracing, check out the [API documentation](https://github.com/DataDog/dd-trace-rb/blob/master/docs/GettingStarted.md#distributed-tracing).
+
+#### Processing pipeline
+
+The processing pipeline allows you to modify traces before they are sent to the agent. Useful for customizing trace content or removing unwanted traces.
```
Comment (lejulie) [inline content/tracing/setup/ruby.md]: "This can be useful for customizing trace content..."

File: content/tracing/setup/ruby.md (hunk 3 above)
Comment (delner) [reply content/tracing/setup/ruby.md]: Because the visual difference between H4 and H3 is almost indiscernible.

**Comment by delner** [comment]
This is the complementary pull request for our `dd-trace-rb` repository: https://github.com/DataDog/dd-trace-rb/pull/388

**Comment by delner** [comment]
Awesome, thanks guys.

**Pull request merged**